packages/xdefi/src/index.ts

```diff
@@ -0,0 +1,35 @@
+import type { WalletInit } from '@web3-onboard/common'
+import { createEIP1193Provider } from '@web3-onboard/common'
+import { CustomWindow } from './types.js'
+declare const window: CustomWindow
+
+function XDEFIWallet(): WalletInit {
+  if (typeof window === 'undefined') return () => null
+  return () => {
+    return {
+      label: 'XDEFI Wallet',
+      injectedNamespace: 'xfi',
+      checkProviderIdentity: ({ provider }: { provider: any }) => {
+        !!provider && !!provider['isXDEFI']
+      },
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async () => {
+        const provider = window.xfi?.ethereum
```

Comment:
Unfortunately for compatibility issues across bundlers and compilers we cannot use optional chaining.
```suggestion
       let provider
        if (window.xfi && window.xfi.ethereum) {
          provider = window.xfi.ethereum
        }
```